Subject: [PATCH] fix medfwrap symbols
---
Index: src/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
--- a/src/CMakeLists.txt	(revision 5a8ca1686bb941ed34018001db7e4d86a3b1bfe7)
+++ b/src/CMakeLists.txt	(date 1713957340468)
@@ -114,7 +114,18 @@
   IF(MEDFILE_BUILD_SHARED_LIBS)
     ## Intermediate library - this is unavoidable 
     ## if we want the Fortran wrapper to build correctly under win.
-    ADD_LIBRARY(medfwrap SHARED ${medfort_wrap_SOURCES})
+    if (MSVC)
+      # Due to issue https://gitlab.kitware.com/cmake/cmake/-/issues/25598
+      # I need to manually set the target link option (otherwise I would have to use intel compiler)
+      configure_file (medfwrap_symbols.def.in medfwrap_symbols.def @ONLY)
+      set (DLLDEF "${CMAKE_CURRENT_BINARY_DIR}/medfwrap_symbols.def")
+      ADD_LIBRARY(medfwrap SHARED ${medfort_wrap_SOURCES})
+      target_link_options(medfwrap PRIVATE "/DEF:${DLLDEF}")
+    else ()
+      ADD_LIBRARY(medfwrap SHARED ${medfort_wrap_SOURCES})
+    endif ()
+    # Add a particular link flag for medfwrap.dll
+
     SET_TARGET_PROPERTIES(medfwrap PROPERTIES
       SOVERSION 11
       VERSION   11.1.1)