diff --git a/config/cmake_files/medMacros.cmake b/config/cmake_files/medMacros.cmake
index 79698ad..ca4350b 100644
--- a/config/cmake_files/medMacros.cmake
+++ b/config/cmake_files/medMacros.cmake
@@ -444,16 +444,11 @@ MACRO(MED_FIND_HDF5)
     
     FIND_PACKAGE(MedfileHDF5 REQUIRED)
 
-    ##
-    ## Requires 1.10.x version
-    ##
-    IF (NOT HDF_VERSION_MAJOR_REF EQUAL 1 OR NOT HDF_VERSION_MINOR_REF EQUAL 10 OR NOT HDF_VERSION_RELEASE_REF GREATER 1)
-        MESSAGE(FATAL_ERROR "HDF5 version is ${HDF_VERSION_REF}. Only versions >= 1.10.2 are supported.")
-    ENDIF()
-    ##
-    ##
-
     ADD_DEFINITIONS(-DH5_USE_16_API)  
+    ADD_DEFINITIONS(-DH5O_info_t_vers=1)
+    ADD_DEFINITIONS(-DH5Oget_info_vers=1)
+    ADD_DEFINITIONS(-DH5Oget_info_by_name_vers=1)
+
     IF(WIN32 AND MEDFILE_BUILD_SHARED_LIBS)
       ADD_DEFINITIONS(-D_HDF5USEDLL_ -DH5_BUILT_AS_DYNAMIC_LIB=1)   
     ENDIF()
