diff --git a/Make.inc b/Make.inc
index f99eb8f..bfe8db8 100644
--- a/Make.inc
+++ b/Make.inc
@@ -11,8 +11,8 @@
 #
 # for solaris
 #
-# CC = gcc
-  CC = /usr/lang-4.0/bin/cc
+CC = gcc
+#  CC = /usr/lang-4.0/bin/cc
 #
 # for sgi
 #
@@ -25,8 +25,8 @@
 #----------------------------------------
 #
 #  set the compiler flags
-# 
-# OPTLEVEL = 
+#
+# OPTLEVEL =
 # OPTLEVEL = -g -v
   OPTLEVEL = -O
 # OPTLEVEL = -xO5 -v
@@ -58,13 +58,13 @@
 #
 #  set the purify environment (a memory monitoring tool)
 #
-  PURIFY = 
+  PURIFY =
 # PURIFY = /usr/local/purify-4.0.1/purify
 #
 #  purify wouldn't work with the newest version of the gcc library,
 #  so we had to force loading the old version
 #
-  PURIFY_GCC_VERSION = 
+  PURIFY_GCC_VERSION =
 # PURIFY_GCC_VERSION = -V 2.7.2
 #
 #---------------------------------------------------------------------
@@ -86,14 +86,14 @@
 #
 #  set suffix rule *.c --> *.o
 #
-.c.o : 
+.c.o :
 	$(PURIFY) $(CC) -c $(CFLAGS) $<
 #
 #---------------------------------------------------------------------
 #
 #  set suffix rule *.c --> *.a
 #
-.c.a : 
+.c.a :
 	$(PURIFY) $(CC) -c $(CFLAGS) $<
 	$(AR) $(ARFLAGS) $@ $*.o
 	rm -f $*.o
@@ -102,7 +102,7 @@
 #
 #  MPI install library
 #
-# MPI_INSTALL_DIR = 
+# MPI_INSTALL_DIR =
   MPI_INSTALL_DIR = /usr/local/mpich-1.0.13
 #
 #---------------------------------------------------------------------
@@ -111,7 +111,7 @@
 #
 # for sgi
 #
-# MPI_LIB_PATH = 
+# MPI_LIB_PATH =
 #
 # for solaris
 #
@@ -119,11 +119,11 @@
 #
 # for hp
 #
-# MPI_LIB_PATH = 
+# MPI_LIB_PATH =
 #
 #---------------------------------------------------------------------
 #
-#  MPI libraries 
+#  MPI libraries
 #
 # for solaris
 #
@@ -131,17 +131,17 @@
 #
 # for sgi
 #
-# MPI_LIBS = $(MPI_LIB_PATH) -lmpi -lpthread 
+# MPI_LIBS = $(MPI_LIB_PATH) -lmpi -lpthread
 #
 # for hp
 # MPI_LIBS = -lpthread
-# MPI_LIBS = $(MPI_LIB_PATH) -lpthread 
+# MPI_LIBS = $(MPI_LIB_PATH) -lpthread
 #
 #---------------------------------------------------------------------
 #
 #  MPI include path
 #
-# MPI_INCLUDE_DIR = 
+# MPI_INCLUDE_DIR =
   MPI_INCLUDE_DIR = -I$(MPI_INSTALL_DIR)/include
 #
 #---------------------------------------------------------------------
