Subject: [PATCH] add a py postprocessor
---
Index: CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	(revision cbdf7fc5d5bd5471a55459e92fc8b5fd1f16f651)
+++ b/CMakeLists.txt	(date 1717162059563)
@@ -126,6 +126,20 @@
 
 message(VERBOSE "MUMPS ${MUMPS_UPSTREAM_VERSION} source directory: ${mumps_SOURCE_DIR}")
 
+# Run a python script (for postprocessing mumps src files before building) if PY_POSTPROCESSING_SCRIPT_FILE is defined
+if (DEFINED PY_POSTPROCESSING_SCRIPT_FILE)
+    find_package(Python COMPONENTS Interpreter REQUIRED)
+    execute_process(
+            COMMAND ${Python_EXECUTABLE} ${PY_POSTPROCESSING_SCRIPT_FILE}
+            WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+            RESULT_VARIABLE result
+    )
+    # Check if the Python script executed successfully
+    if (NOT result EQUAL 0)
+        message(FATAL_ERROR "Python script execution failed with result: ${result}")
+    endif ()
+endif ()
+
 # --- MUMPS build
 
 # --- mpiseq (No MPI)
