Subject: [PATCH] remove vtk-functions
---
Index: tests/test_jupyter.py
===================================================================
diff --git a/tests/test_jupyter.py b/tests/test_jupyter.py
deleted file mode 100644
--- a/tests/test_jupyter.py	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
+++ /dev/null	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
@@ -1,27 +0,0 @@
-from tests import BaseTest
-
-import cadquery as cq
-from cadquery.occ_impl.jupyter_tools import display
-
-
-class TestJupyter(BaseTest):
-    def test_repr_javascript(self):
-        cube = cq.Workplane("XY").box(1, 1, 1)
-        assy = cq.Assembly().add(cube)
-        shape = cube.val()
-
-        self.assertIsInstance(shape, cq.occ_impl.shapes.Solid)
-
-        # Test no exception on rendering to js
-        js1 = shape._repr_javascript_()
-        js2 = cube._repr_javascript_()
-        js3 = assy._repr_javascript_()
-
-        assert "function render" in js1
-        assert "function render" in js2
-        assert "function render" in js3
-
-    def test_display_error(self):
-
-        with self.assertRaises(ValueError):
-            display(cq.Vector())
Index: tests/test_vis.py
===================================================================
diff --git a/tests/test_vis.py b/tests/test_vis.py
deleted file mode 100644
--- a/tests/test_vis.py	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
+++ /dev/null	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
@@ -1,94 +0,0 @@
-from cadquery import Workplane, Assembly, Sketch, Location, Vector
-from cadquery.vis import show, show_object, vtkAxesActor
-
-import cadquery.vis as vis
-
-from vtkmodules.vtkRenderingCore import vtkRenderWindow, vtkRenderWindowInteractor
-from vtkmodules.vtkRenderingAnnotation import vtkAnnotatedCubeActor
-
-from pytest import fixture
-
-
-@fixture
-def wp():
-
-    return Workplane().box(1, 1, 1)
-
-
-@fixture
-def assy(wp):
-
-    return Assembly().add(wp)
-
-
-@fixture
-def sk():
-
-    return Sketch().circle(1.0)
-
-
-class FakeInteractor(vtkRenderWindowInteractor):
-    def Start(self):
-
-        pass
-
-    def Initialize(self):
-
-        pass
-
-
-class FakeWindow(vtkRenderWindow):
-    def Render(*args):
-
-        pass
-
-    def SetSize(*args):
-
-        pass
-
-    def GetScreenSize(*args):
-
-        return 1, 1
-
-    def SetPosition(*args):
-
-        pass
-
-
-def test_show(wp, assy, sk, monkeypatch):
-
-    # use some dummy vtk objects
-    monkeypatch.setattr(vis, "vtkRenderWindowInteractor", FakeInteractor)
-    monkeypatch.setattr(vis, "vtkRenderWindow", FakeWindow)
-
-    # simple smoke test
-    show(wp)
-    show(wp.val())
-    show(wp.val().wrapped)
-    show(assy)
-    show(sk)
-    show(wp, sk, assy, wp.val())
-    show(Vector())
-    show(Location())
-    show([Vector, Vector, Location])
-    show([wp, assy])
-    show()
-
-    # show with edges
-    show(wp, edges=True)
-
-    show_object(wp)
-    show_object(wp.val())
-    show_object(assy)
-    show_object(sk)
-    show_object(wp, sk, assy, wp.val())
-    show_object()
-
-    # for compatibility with CQ-editor
-    show_object(wp, "a")
-
-    # for now a workaround to be compatible with more complicated CQ-editor invocations
-    show(1)
-
-    # show a raw vtkProp
-    show(vtkAxesActor(), [vtkAnnotatedCubeActor()])
Index: environment.yml
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/environment.yml b/environment.yml
--- a/environment.yml	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
+++ b/environment.yml	(date 1738001537689)
@@ -1,11 +1,11 @@
 name: cadquery
 channels:
+  - C:\Work\code\condapackaging\output
   - conda-forge
 dependencies:
   - python>=3.9,<=3.12
   - ipython
-  - ocp=7.8.1
-  - vtk=*=qt*
+  - ocp=7.8.1.1=*novtk*
   - pyparsing>=2.1.9
   - sphinx=8.1.3
   - sphinx_rtd_theme
Index: cadquery/cq_directive.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/cadquery/cq_directive.py b/cadquery/cq_directive.py
--- a/cadquery/cq_directive.py	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
+++ b/cadquery/cq_directive.py	(date 1738001997701)
@@ -9,7 +9,6 @@
 
 from cadquery import exporters, Assembly, Compound, Color, Sketch
 from cadquery import cqgi
-from cadquery.occ_impl.assembly import toJSON
 from cadquery.occ_impl.jupyter_tools import DEFAULT_COLOR
 from docutils.parsers.rst import directives, Directive
 
Index: cadquery/occ_impl/shapes.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/cadquery/occ_impl/shapes.py b/cadquery/occ_impl/shapes.py
--- a/cadquery/occ_impl/shapes.py	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
+++ b/cadquery/occ_impl/shapes.py	(date 1738000682600)
@@ -17,8 +17,8 @@
 
 from io import BytesIO
 
-from vtkmodules.vtkCommonDataModel import vtkPolyData
-from vtkmodules.vtkFiltersCore import vtkTriangleFilter, vtkPolyDataNormals
+# from vtkmodules.vtkCommonDataModel import vtkPolyData
+# from vtkmodules.vtkFiltersCore import vtkTriangleFilter, vtkPolyDataNormals
 
 from .geom import Vector, VectorLike, BoundBox, Plane, Location, Matrix
 from .shape_protocols import geom_LUT_FACE, geom_LUT_EDGE, Shapes, Geoms
@@ -261,8 +261,8 @@
 from OCP.BRepProj import BRepProj_Projection
 from OCP.BRepExtrema import BRepExtrema_DistShapeShape
 
-from OCP.IVtkOCC import IVtkOCC_Shape, IVtkOCC_ShapeMesher
-from OCP.IVtkVTK import IVtkVTK_ShapeData
+# from OCP.IVtkOCC import IVtkOCC_Shape, IVtkOCC_ShapeMesher
+# from OCP.IVtkVTK import IVtkVTK_ShapeData
 
 # for catching exceptions
 from OCP.Standard import Standard_NoSuchObject, Standard_Failure
@@ -1484,53 +1484,53 @@
 
         return self.__class__(result)
 
-    def toVtkPolyData(
-        self,
-        tolerance: Optional[float] = None,
-        angularTolerance: Optional[float] = None,
-        normals: bool = False,
-    ) -> vtkPolyData:
-        """
-        Convert shape to vtkPolyData
-        """
-
-        vtk_shape = IVtkOCC_Shape(self.wrapped)
-        shape_data = IVtkVTK_ShapeData()
-        shape_mesher = IVtkOCC_ShapeMesher()
-
-        drawer = vtk_shape.Attributes()
-        drawer.SetUIsoAspect(Prs3d_IsoAspect(Quantity_Color(), Aspect_TOL_SOLID, 1, 0))
-        drawer.SetVIsoAspect(Prs3d_IsoAspect(Quantity_Color(), Aspect_TOL_SOLID, 1, 0))
-
-        if tolerance:
-            drawer.SetDeviationCoefficient(tolerance)
-
-        if angularTolerance:
-            drawer.SetDeviationAngle(angularTolerance)
-
-        shape_mesher.Build(vtk_shape, shape_data)
-
-        rv = shape_data.getVtkPolyData()
-
-        # convert to triangles and split edges
-        t_filter = vtkTriangleFilter()
-        t_filter.SetInputData(rv)
-        t_filter.Update()
-
-        rv = t_filter.GetOutput()
-
-        # compute normals
-        if normals:
-            n_filter = vtkPolyDataNormals()
-            n_filter.SetComputePointNormals(True)
-            n_filter.SetComputeCellNormals(True)
-            n_filter.SetFeatureAngle(360)
-            n_filter.SetInputData(rv)
-            n_filter.Update()
-
-            rv = n_filter.GetOutput()
-
-        return rv
+    # def toVtkPolyData(
+    #     self,
+    #     tolerance: Optional[float] = None,
+    #     angularTolerance: Optional[float] = None,
+    #     normals: bool = False,
+    # ) -> vtkPolyData:
+    #     """
+    #     Convert shape to vtkPolyData
+    #     """
+    #
+    #     vtk_shape = IVtkOCC_Shape(self.wrapped)
+    #     shape_data = IVtkVTK_ShapeData()
+    #     shape_mesher = IVtkOCC_ShapeMesher()
+    #
+    #     drawer = vtk_shape.Attributes()
+    #     drawer.SetUIsoAspect(Prs3d_IsoAspect(Quantity_Color(), Aspect_TOL_SOLID, 1, 0))
+    #     drawer.SetVIsoAspect(Prs3d_IsoAspect(Quantity_Color(), Aspect_TOL_SOLID, 1, 0))
+    #
+    #     if tolerance:
+    #         drawer.SetDeviationCoefficient(tolerance)
+    #
+    #     if angularTolerance:
+    #         drawer.SetDeviationAngle(angularTolerance)
+    #
+    #     shape_mesher.Build(vtk_shape, shape_data)
+    #
+    #     rv = shape_data.getVtkPolyData()
+    #
+    #     # convert to triangles and split edges
+    #     t_filter = vtkTriangleFilter()
+    #     t_filter.SetInputData(rv)
+    #     t_filter.Update()
+    #
+    #     rv = t_filter.GetOutput()
+    #
+    #     # compute normals
+    #     if normals:
+    #         n_filter = vtkPolyDataNormals()
+    #         n_filter.SetComputePointNormals(True)
+    #         n_filter.SetComputeCellNormals(True)
+    #         n_filter.SetFeatureAngle(360)
+    #         n_filter.SetInputData(rv)
+    #         n_filter.Update()
+    #
+    #         rv = n_filter.GetOutput()
+    #
+    #     return rv
 
     def _repr_javascript_(self):
         """
Index: tests/test_exporters.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tests/test_exporters.py b/tests/test_exporters.py
--- a/tests/test_exporters.py	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
+++ b/tests/test_exporters.py	(date 1738002145028)
@@ -629,27 +629,6 @@
 
         exporters.export(self._box(), "out.tjs")
 
-    def testVRML(self):
-
-        exporters.export(self._box(), "out.vrml")
-
-        with open("out.vrml") as f:
-            res = f.read(10)
-
-        assert res.startswith("#VRML V2.0")
-
-        # export again to trigger all paths in the code
-        exporters.export(self._box(), "out.vrml")
-
-    def testVTP(self):
-
-        exporters.export(self._box(), "out.vtp")
-
-        with open("out.vtp") as f:
-            res = f.read(100)
-
-        assert res.startswith('<?xml version="1.0"?>\n<VTKFile')
-
     def testDXF(self):
 
         exporters.export(self._box().section(), "out.dxf")
@@ -823,32 +802,6 @@
         r = f.read(len(matchval))
         assert r == matchval
 
-
-def test_assy_vtk_rotation(tmpdir):
-
-    v0 = Vertex.makeVertex(1, 0, 0)
-
-    assy = Assembly()
-    assy.add(
-        v0, name="v0", loc=Location(Vector(0, 0, 0), Vector(1, 0, 0), 90),
-    )
-
-    fwrl = Path(tmpdir) / "v0.wrl"
-    assert not fwrl.exists()
-    assy.save(str(fwrl), "VRML")
-    assert fwrl.exists()
-
-    matched_rot = False
-    with open(fwrl) as f:
-        pat_rot = re.compile("""rotation 1 0 0 1.5707963267""")
-        for line in f:
-            if m := re.search(pat_rot, line):
-                matched_rot = True
-                break
-
-    assert matched_rot
-
-
 def test_tessellate(box123):
 
     verts, triangles = box123.val().tessellate(1e-6)
Index: cadquery/occ_impl/exporters/assembly.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/cadquery/occ_impl/exporters/assembly.py b/cadquery/occ_impl/exporters/assembly.py
--- a/cadquery/occ_impl/exporters/assembly.py	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
+++ b/cadquery/occ_impl/exporters/assembly.py	(date 1738000682574)
@@ -7,8 +7,8 @@
 from typing import Optional
 from typing_extensions import Literal
 
-from vtkmodules.vtkIOExport import vtkJSONSceneExporter, vtkVRMLExporter
-from vtkmodules.vtkRenderingCore import vtkRenderer, vtkRenderWindow
+# from vtkmodules.vtkIOExport import vtkJSONSceneExporter, vtkVRMLExporter
+# from vtkmodules.vtkRenderingCore import vtkRenderer, vtkRenderWindow
 
 from OCP.XSControl import XSControl_WorkSession
 from OCP.STEPCAFControl import STEPCAFControl_Writer
@@ -26,7 +26,7 @@
 from OCP.Message import Message_ProgressRange
 from OCP.Interface import Interface_Static
 
-from ..assembly import AssemblyProtocol, toCAF, toVTK, toFusedCAF
+from ..assembly import AssemblyProtocol, toCAF, toFusedCAF
 from ..geom import Location
 
 
@@ -134,53 +134,53 @@
     return status == PCDM_StoreStatus.PCDM_SS_OK
 
 
-def _vtkRenderWindow(
-    assy: AssemblyProtocol, tolerance: float = 1e-3, angularTolerance: float = 0.1
-) -> vtkRenderWindow:
-    """
-    Convert an assembly to a vtkRenderWindow. Used by vtk based exporters.
-    """
-
-    renderer = toVTK(assy, tolerance=tolerance, angularTolerance=angularTolerance)
-    renderWindow = vtkRenderWindow()
-    renderWindow.AddRenderer(renderer)
-
-    renderer.ResetCamera()
-    renderer.SetBackground(1, 1, 1)
-
-    return renderWindow
+# def _vtkRenderWindow(
+#     assy: AssemblyProtocol, tolerance: float = 1e-3, angularTolerance: float = 0.1
+# ) -> vtkRenderWindow:
+#     """
+#     Convert an assembly to a vtkRenderWindow. Used by vtk based exporters.
+#     """
+#
+#     renderer = toVTK(assy, tolerance=tolerance, angularTolerance=angularTolerance)
+#     renderWindow = vtkRenderWindow()
+#     renderWindow.AddRenderer(renderer)
+#
+#     renderer.ResetCamera()
+#     renderer.SetBackground(1, 1, 1)
+#
+#     return renderWindow
 
 
-def exportVTKJS(assy: AssemblyProtocol, path: str):
-    """
-    Export an assembly to a zipped vtkjs. NB: .zip extensions is added to path.
-    """
+# def exportVTKJS(assy: AssemblyProtocol, path: str):
+#     """
+#     Export an assembly to a zipped vtkjs. NB: .zip extensions is added to path.
+#     """
+#
+#     renderWindow = _vtkRenderWindow(assy)
+#
+#     with TemporaryDirectory() as tmpdir:
+#
+#         exporter = vtkJSONSceneExporter()
+#         exporter.SetFileName(tmpdir)
+#         exporter.SetRenderWindow(renderWindow)
+#         exporter.Write()
+#         make_archive(path, "zip", tmpdir)
 
-    renderWindow = _vtkRenderWindow(assy)
 
-    with TemporaryDirectory() as tmpdir:
-
-        exporter = vtkJSONSceneExporter()
-        exporter.SetFileName(tmpdir)
-        exporter.SetRenderWindow(renderWindow)
-        exporter.Write()
-        make_archive(path, "zip", tmpdir)
-
-
-def exportVRML(
-    assy: AssemblyProtocol,
-    path: str,
-    tolerance: float = 1e-3,
-    angularTolerance: float = 0.1,
-):
-    """
-    Export an assembly to a vrml file using vtk.
-    """
-
-    exporter = vtkVRMLExporter()
-    exporter.SetFileName(path)
-    exporter.SetRenderWindow(_vtkRenderWindow(assy, tolerance, angularTolerance))
-    exporter.Write()
+# def exportVRML(
+#     assy: AssemblyProtocol,
+#     path: str,
+#     tolerance: float = 1e-3,
+#     angularTolerance: float = 0.1,
+# ):
+#     """
+#     Export an assembly to a vrml file using vtk.
+#     """
+#
+#     exporter = vtkVRMLExporter()
+#     exporter.SetFileName(path)
+#     exporter.SetRenderWindow(_vtkRenderWindow(assy, tolerance, angularTolerance))
+#     exporter.Write()
 
 
 def exportGLTF(
Index: tests/test_assembly.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tests/test_assembly.py b/tests/test_assembly.py
--- a/tests/test_assembly.py	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
+++ b/tests/test_assembly.py	(date 1738001248334)
@@ -12,10 +12,8 @@
 from cadquery.occ_impl.exporters.assembly import (
     exportAssembly,
     exportCAF,
-    exportVTKJS,
-    exportVRML,
 )
-from cadquery.occ_impl.assembly import toJSON, toCAF, toFusedCAF
+from cadquery.occ_impl.assembly import toCAF, toFusedCAF
 from cadquery.occ_impl.shapes import Face, box
 
 from OCP.gp import gp_XYZ
@@ -641,34 +639,6 @@
     # only sanity check for now
     assert os.path.exists("assy.xml")
 
-
-def test_vtkjs_export(nested_assy):
-
-    exportVTKJS(nested_assy, "assy")
-
-    # only sanity check for now
-    assert os.path.exists("assy.zip")
-
-
-def test_vrml_export(simple_assy):
-
-    exportVRML(simple_assy, "assy.wrl")
-
-    # only sanity check for now
-    assert os.path.exists("assy.wrl")
-
-
-def test_toJSON(simple_assy, nested_assy, empty_top_assy):
-
-    r1 = toJSON(simple_assy)
-    r2 = toJSON(simple_assy)
-    r3 = toJSON(empty_top_assy)
-
-    assert len(r1) == 3
-    assert len(r2) == 3
-    assert len(r3) == 1
-
-
 @pytest.mark.parametrize(
     "extension, args",
     [
@@ -676,7 +646,6 @@
         ("xml", ()),
         ("stp", ("STEP",)),
         ("caf", ("XML",)),
-        ("wrl", ("VRML",)),
         ("stl", ("STL",)),
     ],
 )
@@ -692,14 +661,12 @@
     [
         ("step", (), {}),
         ("xml", (), {}),
-        ("vrml", (), {}),
         ("gltf", (), {}),
         ("glb", (), {}),
         ("stl", (), {"ascii": False}),
         ("stl", (), {"ascii": True}),
         ("stp", ("STEP",), {}),
         ("caf", ("XML",), {}),
-        ("wrl", ("VRML",), {}),
         ("stl", ("STL",), {}),
     ],
 )
@@ -712,12 +679,6 @@
         assert os.path.exists(filename)
 
 
-def test_export_vtkjs(tmpdir, nested_assy):
-
-    with tmpdir:
-        nested_assy.export("nested.vtkjs")
-        assert os.path.exists("nested.vtkjs.zip")
-
 
 def test_export_errors(nested_assy):
 
@@ -805,12 +766,6 @@
     assert output.err == ""
 
 
-def test_save_vtkjs(nested_assy):
-
-    nested_assy.save("nested", "VTKJS")
-    assert os.path.exists("nested.zip")
-
-
 def test_save_raises(nested_assy):
 
     with pytest.raises(ValueError):
Index: cadquery/occ_impl/assembly.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/cadquery/occ_impl/assembly.py b/cadquery/occ_impl/assembly.py
--- a/cadquery/occ_impl/assembly.py	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
+++ b/cadquery/occ_impl/assembly.py	(date 1738000682581)
@@ -27,18 +27,18 @@
 from OCP.TopoDS import TopoDS_Shape
 from OCP.gp import gp_EulerSequence
 
-from vtkmodules.vtkRenderingCore import (
-    vtkActor,
-    vtkPolyDataMapper as vtkMapper,
-    vtkRenderer,
-)
-
-from vtkmodules.vtkFiltersExtraction import vtkExtractCellsByType
-from vtkmodules.vtkCommonDataModel import VTK_TRIANGLE, VTK_LINE, VTK_VERTEX
+# from vtkmodules.vtkRenderingCore import (
+#     vtkActor,
+#     vtkPolyDataMapper as vtkMapper,
+#     vtkRenderer,
+# )
+#
+# from vtkmodules.vtkFiltersExtraction import vtkExtractCellsByType
+# from vtkmodules.vtkCommonDataModel import VTK_TRIANGLE, VTK_LINE, VTK_VERTEX
 
 from .geom import Location
 from .shapes import Shape, Solid, Compound
-from .exporters.vtk import toString
+# from .exporters.vtk import toString
 from ..cq import Workplane
 
 # type definitions
@@ -278,97 +278,97 @@
     return trans, (rot[1], rot[2], rot[0])
 
 
-def toVTK(
-    assy: AssemblyProtocol,
-    color: Tuple[float, float, float, float] = (1.0, 1.0, 1.0, 1.0),
-    tolerance: float = 1e-3,
-    angularTolerance: float = 0.1,
-) -> vtkRenderer:
-
-    renderer = vtkRenderer()
-
-    for shape, _, loc, col_ in assy:
-
-        col = col_.toTuple() if col_ else color
-
-        trans, rot = _loc2vtk(loc)
-
-        data = shape.toVtkPolyData(tolerance, angularTolerance)
-
-        # extract faces
-        extr = vtkExtractCellsByType()
-        extr.SetInputDataObject(data)
-
-        extr.AddCellType(VTK_LINE)
-        extr.AddCellType(VTK_VERTEX)
-        extr.Update()
-        data_edges = extr.GetOutput()
-
-        # extract edges
-        extr = vtkExtractCellsByType()
-        extr.SetInputDataObject(data)
-
-        extr.AddCellType(VTK_TRIANGLE)
-        extr.Update()
-        data_faces = extr.GetOutput()
-
-        # remove normals from edges
-        data_edges.GetPointData().RemoveArray("Normals")
-
-        # add both to the renderer
-        mapper = vtkMapper()
-        mapper.AddInputDataObject(data_faces)
-
-        actor = vtkActor()
-        actor.SetMapper(mapper)
-        actor.SetPosition(*trans)
-        actor.SetOrientation(*rot)
-        actor.GetProperty().SetColor(*col[:3])
-        actor.GetProperty().SetOpacity(col[3])
-
-        renderer.AddActor(actor)
-
-        mapper = vtkMapper()
-        mapper.AddInputDataObject(data_edges)
-
-        actor = vtkActor()
-        actor.SetMapper(mapper)
-        actor.SetPosition(*trans)
-        actor.SetOrientation(*rot)
-        actor.GetProperty().SetColor(0, 0, 0)
-        actor.GetProperty().SetLineWidth(2)
-
-        renderer.AddActor(actor)
-
-    return renderer
-
-
-def toJSON(
-    assy: AssemblyProtocol,
-    color: Tuple[float, float, float, float] = (1.0, 1.0, 1.0, 1.0),
-    tolerance: float = 1e-3,
-) -> List[Dict[str, Any]]:
-    """
-    Export an object to a structure suitable for converting to VTK.js JSON.
-    """
-
-    rv = []
-
-    for shape, _, loc, col_ in assy:
-
-        val: Any = {}
+# def toVTK(
+#     assy: AssemblyProtocol,
+#     color: Tuple[float, float, float, float] = (1.0, 1.0, 1.0, 1.0),
+#     tolerance: float = 1e-3,
+#     angularTolerance: float = 0.1,
+# ) -> vtkRenderer:
+#
+#     renderer = vtkRenderer()
+#
+#     for shape, _, loc, col_ in assy:
+#
+#         col = col_.toTuple() if col_ else color
+#
+#         trans, rot = _loc2vtk(loc)
+#
+#         data = shape.toVtkPolyData(tolerance, angularTolerance)
+#
+#         # extract faces
+#         extr = vtkExtractCellsByType()
+#         extr.SetInputDataObject(data)
+#
+#         extr.AddCellType(VTK_LINE)
+#         extr.AddCellType(VTK_VERTEX)
+#         extr.Update()
+#         data_edges = extr.GetOutput()
+#
+#         # extract edges
+#         extr = vtkExtractCellsByType()
+#         extr.SetInputDataObject(data)
+#
+#         extr.AddCellType(VTK_TRIANGLE)
+#         extr.Update()
+#         data_faces = extr.GetOutput()
+#
+#         # remove normals from edges
+#         data_edges.GetPointData().RemoveArray("Normals")
+#
+#         # add both to the renderer
+#         mapper = vtkMapper()
+#         mapper.AddInputDataObject(data_faces)
+#
+#         actor = vtkActor()
+#         actor.SetMapper(mapper)
+#         actor.SetPosition(*trans)
+#         actor.SetOrientation(*rot)
+#         actor.GetProperty().SetColor(*col[:3])
+#         actor.GetProperty().SetOpacity(col[3])
+#
+#         renderer.AddActor(actor)
+#
+#         mapper = vtkMapper()
+#         mapper.AddInputDataObject(data_edges)
+#
+#         actor = vtkActor()
+#         actor.SetMapper(mapper)
+#         actor.SetPosition(*trans)
+#         actor.SetOrientation(*rot)
+#         actor.GetProperty().SetColor(0, 0, 0)
+#         actor.GetProperty().SetLineWidth(2)
+#
+#         renderer.AddActor(actor)
+#
+#     return renderer
 
-        data = toString(shape, tolerance)
-        trans, rot = loc.toTuple()
 
-        val["shape"] = data
-        val["color"] = col_.toTuple() if col_ else color
-        val["position"] = trans
-        val["orientation"] = tuple(radians(r) for r in rot)
-
-        rv.append(val)
-
-    return rv
+# def toJSON(
+#     assy: AssemblyProtocol,
+#     color: Tuple[float, float, float, float] = (1.0, 1.0, 1.0, 1.0),
+#     tolerance: float = 1e-3,
+# ) -> List[Dict[str, Any]]:
+#     """
+#     Export an object to a structure suitable for converting to VTK.js JSON.
+#     """
+#
+#     rv = []
+#
+#     for shape, _, loc, col_ in assy:
+#
+#         val: Any = {}
+#
+#         data = toString(shape, tolerance)
+#         trans, rot = loc.toTuple()
+#
+#         val["shape"] = data
+#         val["color"] = col_.toTuple() if col_ else color
+#         val["position"] = trans
+#         val["orientation"] = tuple(radians(r) for r in rot)
+#
+#         rv.append(val)
+#
+#     return rv
 
 
 def toFusedCAF(
Index: cadquery/assembly.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/cadquery/assembly.py b/cadquery/assembly.py
--- a/cadquery/assembly.py	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
+++ b/cadquery/assembly.py	(date 1738000682608)
@@ -29,8 +29,6 @@
 from .occ_impl.exporters.assembly import (
     exportAssembly,
     exportCAF,
-    exportVTKJS,
-    exportVRML,
     exportGLTF,
     STEPExportModeLiterals,
 )
@@ -491,11 +489,13 @@
         elif exportType == "XML":
             exportCAF(self, path)
         elif exportType == "VRML":
-            exportVRML(self, path, tolerance, angularTolerance)
+            # exportVRML(self, path, tolerance, angularTolerance)
+            raise Exception("You are running NOVTK. VRML export is not supported")
         elif exportType == "GLTF" or exportType == "GLB":
             exportGLTF(self, path, None, tolerance, angularTolerance)
         elif exportType == "VTKJS":
-            exportVTKJS(self, path)
+            # exportVTKJS(self, path)
+            raise Exception("You are running NOVTK. VRML export is not supported")
         elif exportType == "STL":
             # Handle the ascii setting for STL export
             export_ascii = False
Index: cadquery/occ_impl/exporters/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/cadquery/occ_impl/exporters/__init__.py b/cadquery/occ_impl/exporters/__init__.py
--- a/cadquery/occ_impl/exporters/__init__.py	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
+++ b/cadquery/occ_impl/exporters/__init__.py	(date 1738000682566)
@@ -15,7 +15,7 @@
 from .amf import AmfWriter
 from .threemf import ThreeMFWriter
 from .dxf import exportDXF, DxfDocument
-from .vtk import exportVTP
+# from .vtk import exportVTP
 
 
 class ExportTypes:
@@ -124,7 +124,8 @@
         VrmlAPI.Write_s(shape.wrapped, fname)
 
     elif exportType == ExportTypes.VTP:
-        exportVTP(shape, fname, tolerance, angularTolerance)
+        # exportVTP(shape, fname, tolerance, angularTolerance)
+        raise NotImplementedError("You are using the cadquery NOVTK build, VTP export is not supported")
 
     elif exportType == ExportTypes.BREP:
         shape.exportBrep(fname)
Index: tests/test_cadquery.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tests/test_cadquery.py b/tests/test_cadquery.py
--- a/tests/test_cadquery.py	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
+++ b/tests/test_cadquery.py	(date 1738000682623)
@@ -5731,15 +5731,15 @@
         with raises(ValueError):
             Face.makeNSidedSurface(outer_w, [[0, 0, 1]])
 
-    def test_toVtk(self):
-
-        from vtkmodules.vtkCommonDataModel import vtkPolyData
-
-        f = Face.makePlane(2, 2)
-        vtk = f.toVtkPolyData(normals=False)
-
-        assert isinstance(vtk, vtkPolyData)
-        assert vtk.GetNumberOfPolys() == 2
+    # def test_toVtk(self):
+    #
+    #     from vtkmodules.vtkCommonDataModel import vtkPolyData
+    #
+    #     f = Face.makePlane(2, 2)
+    #     vtk = f.toVtkPolyData(normals=False)
+    #
+    #     assert isinstance(vtk, vtkPolyData)
+    #     assert vtk.GetNumberOfPolys() == 2
 
     def test_iterators(self):
 
Index: cadquery/occ_impl/jupyter_tools.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/cadquery/occ_impl/jupyter_tools.py b/cadquery/occ_impl/jupyter_tools.py
--- a/cadquery/occ_impl/jupyter_tools.py	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
+++ b/cadquery/occ_impl/jupyter_tools.py	(date 1738002067363)
@@ -4,11 +4,11 @@
 from IPython.display import Javascript
 
 
-from .exporters.vtk import toString
+#from .exporters.vtk import toString
 from .shapes import Shape
 from ..assembly import Assembly
-from .assembly import toJSON
-from ..vis import DEFAULT_COLOR
+#from .assembly import toJSON
+#from ..vis import DEFAULT_COLOR
 
 TEMPLATE_RENDER = """
 
@@ -161,7 +161,6 @@
 
 
 def display(shape):
-
     payload: List[Dict[str, Any]] = []
 
     if isinstance(shape, Shape):
Index: .coveragerc
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/.coveragerc b/.coveragerc
--- a/.coveragerc	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
+++ b/.coveragerc	(date 1738002450666)
@@ -3,6 +3,8 @@
 omit = 
     cadquery/utils.py
     cadquery/cq_directive.py
+    cadquery/vis.py
+    cadquery/occ_impl/exporters/vtk.py
     tests/*
 
 [report]
Index: tests/test_examples.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/tests/test_examples.py b/tests/test_examples.py
--- a/tests/test_examples.py	(revision f13f60735ff0593524097eb6ed47f97ef1ee722a)
+++ b/tests/test_examples.py	(date 1738002081791)
@@ -11,7 +11,7 @@
 
 import cadquery as cq
 from cadquery import cqgi
-from cadquery.cq_directive import cq_directive
+#from cadquery.cq_directive import cq_directive
 
 
 def find_examples(pattern="examples/*.py", path=Path("examples")):
@@ -22,39 +22,8 @@
 
         yield code, path
 
-
-def find_examples_in_docs(pattern="doc/*.rst", path=Path("doc")):
-
-    # dummy CQ directive for code
-    class dummy_cq_directive(cq_directive):
-
-        codes = []
-
-        def run(self):
-
-            self.codes.append("\n".join(self.content))
-
-            return []
-
-    directives.register_directive("cadquery", dummy_cq_directive)
-
-    # read and parse all rst files
-    for p in glob(pattern):
-        with open(p, encoding="UTF-8") as f:
-            doc = f.read()
-
-        publish_doctree(
-            doc, settings_overrides={"report_level": Reporter.SEVERE_LEVEL + 1}
-        )
-
-    # yield all code snippets
-    for c in dummy_cq_directive.codes:
-
-        yield c, path
-
-
 @pytest.mark.parametrize(
-    "code, path", chain(find_examples(), find_examples_in_docs()), ids=count(0)
+    "code, path", chain(find_examples()), ids=count(0)
 )
 def test_example(code, path):
 
