Subject: [PATCH] do not use debug suffices
---
Index: adm/cmake/occt_macros.cmake
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/adm/cmake/occt_macros.cmake b/adm/cmake/occt_macros.cmake
--- a/adm/cmake/occt_macros.cmake	(revision 6e70ffc7ad1643adaedce1012ad08275eda4e5e2)
+++ b/adm/cmake/occt_macros.cmake	(date 1725983321743)
@@ -607,9 +607,9 @@
   install(CODE "if (\"\${CMAKE_INSTALL_CONFIG_NAME}\" MATCHES \"^([Rr][Ee][Ll][Ee][Aa][Ss][Ee])$\")
     set (OCCT_INSTALL_BIN_LETTER \"\")
   elseif (\"\${CMAKE_INSTALL_CONFIG_NAME}\" MATCHES \"^([Rr][Ee][Ll][Ww][Ii][Tt][Hh][Dd][Ee][Bb][Ii][Nn][Ff][Oo])$\")
-    set (OCCT_INSTALL_BIN_LETTER \"i\")
+    set (OCCT_INSTALL_BIN_LETTER \"\")
   elseif (\"\${CMAKE_INSTALL_CONFIG_NAME}\" MATCHES \"^([Dd][Ee][Bb][Uu][Gg])$\")
-    set (OCCT_INSTALL_BIN_LETTER \"d\")
+    set (OCCT_INSTALL_BIN_LETTER \"\")
   endif()")
 endmacro()
 
