Subject: [PATCH] fix fortran support
---
Index: config/ifort.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/config/ifort.py b/config/ifort.py
--- a/config/ifort.py	(revision 80f386e27fd2727ce4a881219970e4b54a043ba9)
+++ b/config/ifort.py	(date 1730580447585)
@@ -102,6 +102,7 @@
     ifort_batch_file = pathlib.Path(os.getenv("INTEL_VARS_PATH")+'\\vars.bat')
     ifort_version = os.getenv("INTEL_FORTRAN_VERSION", None)
     if ifort_batch_file.exists() and ifort_version:
+        Logs.info(f"Using ifort batch file: {ifort_batch_file}")
         arch = 'amd64'
         version = ifort_version
         target = 'intel64'
