Subject: [PATCH] numpy fix
---
Index: code_aster/__init__.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/code_aster/__init__.py b/code_aster/__init__.py
--- a/code_aster/__init__.py	(revision c1b3669de07b489765a57e6e1c30d81def31e20f)
+++ b/code_aster/__init__.py	(date 1706948242017)
@@ -64,6 +64,13 @@

 import os

+# To eliminate need for activation scripts during conda runtime
+_CONDA_PREFIX = os.getenv('CONDA_PREFIX', None)
+_CONDA_BUILD = os.getenv('CONDA_BUILD', False)
+if _CONDA_PREFIX is not None and _CONDA_BUILD is False:
+    os.environ['ASTER_ELEMENTSDIR'] = os.getenv('CONDA_PREFIX') + '/lib/aster'
+
+
 try:
     # embedded modules must be imported before libaster
     # because there are set up by initAsterModules
@@ -79,7 +86,8 @@
     # ... except for rc and package info
     from .Utilities.rc import rc
     from .Utilities.version import __version__
-except ImportError:
+except ImportError as e:
+    print(f"ImportErrors: '{e}'")
     # AsterStudy only uses code_aster/Cata without the extensions modules (.so).
     # So, the exception is only raised during the building process.
     if os.environ.get("WAFLOCK"):
