Subject: [PATCH] fix llvm clang win
---
Index: mfront/src/MaterialKnowledgeDescription.cxx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/mfront/src/MaterialKnowledgeDescription.cxx b/mfront/src/MaterialKnowledgeDescription.cxx
--- a/mfront/src/MaterialKnowledgeDescription.cxx	(revision 3b890a212e2bde890cbf12c0c7abf3b3b45a9b29)
+++ b/mfront/src/MaterialKnowledgeDescription.cxx	(date 1710521168536)
@@ -15,6 +15,8 @@
 #include "TFEL/Raise.hxx"
 #include "MFront/MaterialKnowledgeDescription.hxx"
 
+#include <iterator>
+
 namespace mfront {
 
   const char* const MaterialKnowledgeDescription::defaultOutOfBoundsPolicy =
Index: include/TFEL/Utilities/GenTypeBase.hxx
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/include/TFEL/Utilities/GenTypeBase.hxx b/include/TFEL/Utilities/GenTypeBase.hxx
--- a/include/TFEL/Utilities/GenTypeBase.hxx	(revision 3b890a212e2bde890cbf12c0c7abf3b3b45a9b29)
+++ b/include/TFEL/Utilities/GenTypeBase.hxx	(date 1710518808013)
@@ -173,6 +173,10 @@
     TFEL_INLINE type_check<T1, void> set(T1 &&src) {
       this->operator=(std::forward<T1>(src));
     }
+    template <typename T1>
+    TFEL_INLINE void set(const T1 &src) {
+      this->operator=(src);
+    }
     //
     template <typename T1>
     TFEL_INLINE type_check<T1, bool> is() const {
@@ -195,7 +199,7 @@
 
    protected:
     //! \brief clear the GenType
-    TFEL_INLINE void clear() { this->operator=(std::monostate()); }
+    //TFEL_INLINE void clear() { this->operator=(std::monostate()); }
     //! \brief set the value of the GenType.
     template <typename T1>
     TFEL_INLINE type_check<T1, void> set_uninitialised() {
Index: bindings/python/tfel/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bindings/python/tfel/CMakeLists.txt b/bindings/python/tfel/CMakeLists.txt
--- a/bindings/python/tfel/CMakeLists.txt	(revision 3b890a212e2bde890cbf12c0c7abf3b3b45a9b29)
+++ b/bindings/python/tfel/CMakeLists.txt	(date 1710522737934)
@@ -116,6 +116,8 @@
   tfel_python_module(system system.cxx
     LibraryInformation.cxx
     ExternalLibraryManager.cxx
+    ExternalMaterialKnowledgeDescription.cxx
+    ExternalMaterialPropertyDescription.cxx
     ExternalBehaviourDescription.cxx)
 endif(NOT WIN32)
 target_link_libraries(py_tfel_system
