Subject: [PATCH] fix v4.2.0 clang win
---
Index: bindings/python/tfel/CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/bindings/python/tfel/CMakeLists.txt b/bindings/python/tfel/CMakeLists.txt
--- a/bindings/python/tfel/CMakeLists.txt	(revision ccd659c6f5d44488a3a2754e812cbdde436c80dc)
+++ b/bindings/python/tfel/CMakeLists.txt	(date 1710530486951)
@@ -118,6 +118,8 @@
   tfel_python_module(system system.cxx
     LibraryInformation.cxx
     ExternalLibraryManager.cxx
+    ExternalMaterialKnowledgeDescription.cxx
+    ExternalMaterialPropertyDescription.cxx
     ExternalBehaviourDescription.cxx)
 endif(NOT WIN32)
 target_link_libraries(py_tfel_system
