Subject: [PATCH] override site package dir for windows
---
Index: cmake/modules/tfel.cmake
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/cmake/modules/tfel.cmake b/cmake/modules/tfel.cmake
--- a/cmake/modules/tfel.cmake	(revision f22712da02183b8d04eed8124ce4bb393c586085)
+++ b/cmake/modules/tfel.cmake	(date 1710570065230)
@@ -445,7 +445,7 @@
   if(TFEL_APPEND_SUFFIX)
     if(WIN32)
       install(TARGETS py_${package}_${name}
-        DESTINATION bin/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages/${package}_${TFEL_SUFFIX_FOR_PYTHON_MODULES}
+        DESTINATION ${SITE_PACKAGES_DIR}/${package}_${TFEL_SUFFIX_FOR_PYTHON_MODULES}
         COMPONENT python_bindings)
     else(WIN32)
       install(TARGETS py_${package}_${name}
@@ -455,7 +455,7 @@
   else(TFEL_APPEND_SUFFIX)
     if(WIN32)
       install(TARGETS py_${package}_${name}
-        DESTINATION bin/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages/${package}
+        DESTINATION ${SITE_PACKAGES_DIR}/${package}
         COMPONENT python_bindings)
     else(WIN32)
       install(TARGETS py_${package}_${name}        DESTINATION lib${LIB_SUFFIX}/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages/${package}
@@ -520,7 +520,7 @@
     endif(EXISTS "${CMAKE_CURRENT_SOURCE_DIR}/${pyscript}.in")
     if(WIN32)
       install(PROGRAMS ${python_script}
-        DESTINATION bin/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages/${dir}/
+        DESTINATION ${SITE_PACKAGES_DIR}/${dir}/
         COMPONENT python_bindings)
     else(WIN32)
       install(PROGRAMS ${python_script}
Index: CMakeLists.txt
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/CMakeLists.txt b/CMakeLists.txt
--- a/CMakeLists.txt	(revision f22712da02183b8d04eed8124ce4bb393c586085)
+++ b/CMakeLists.txt	(date 1710570288760)
@@ -30,6 +30,13 @@
 set(CMAKE_CXX_STANDARD 17)
 set(CXX_STANDARD_REQUIRED ON)
 
+if(SITE_PACKAGES_DIR)
+  message(STATUS "site-packages directory : ${SITE_PACKAGES_DIR}")
+else ()
+  set(SITE_PACKAGES_DIR "bin/python${PYTHON_VERSION_MAJOR}.${PYTHON_VERSION_MINOR}/site-packages")
+  message(STATUS "site-packages directory : ${SITE_PACKAGES_DIR}")
+endif(SITE_PACKAGES_DIR)
+
 # tfel specific macros
 include(cmake/modules/tfel.cmake)
 # some variables for compatibility with autotools
