Subject: [PATCH] fix hierarchyhelper
---
Index: src/ifcparse/IfcHierarchyHelper.h
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
diff --git a/src/ifcparse/IfcHierarchyHelper.h b/src/ifcparse/IfcHierarchyHelper.h
--- a/src/ifcparse/IfcHierarchyHelper.h	(revision 65179db44405f3e4fb403915950be8cc1e5c328f)
+++ b/src/ifcparse/IfcHierarchyHelper.h	(date 1707933494639)
@@ -436,7 +436,7 @@
 			{ IfcWrite::IfcWriteArgument* attr = new IfcWrite::IfcWriteArgument(); attr->set<std::string>(IfcParse::IfcGlobalId()); data->setArgument(0, attr); }
 			{ IfcWrite::IfcWriteArgument* attr = new IfcWrite::IfcWriteArgument(); attr->set(owner_hist); data->setArgument(1, attr); }
 			int relating_index = 4, related_index = 5;
-			if (T::Class().name() == "IfcRelContainedInSpatialStructure" || std::is_base_of<Schema::IfcRelDefines, T>::value) {
+			if (T::Class().name() == "IfcRelContainedInSpatialStructure" || std::is_base_of<typename Schema::IfcRelDefines, T>::value) {
 				// some classes have attributes reversed.
 				std::swap(relating_index, related_index);
 			}
